Slim/Schema.pm

```diff
@@ -2335,6 +2335,7 @@ sub artistOnlyRoles {
 	my %roles = (
 		'ARTIST'      => 1,
 		'ALBUMARTIST' => 1,
+		'TRACKARTIST' => 1,
```

Comment:
I believe that the sub's name is a bit misleading or dated. "artists only" at the time probably meant, an artist was the main work's artist? Therefore the track artist isn't part of the list? 

I fear that modifying this here might cause unwanted side-effects in eg. 3rd party pluginsns. Why not pass `TRACKARTIST` as part of the `@add` parameter when needed? This way we can make sure we really only get the additional role when we want it. I think there are only one or two callers in the code.